devito/finite_differences/derivative.py

```diff
@@ -115,7 +115,9 @@ def __new__(cls, expr, *dims, **kwargs):
             orders = as_tuple(orders)
 
         # Finite difference orders
-        fd_orders = kwargs.get('fd_order', tuple([expr.space_order]*len(dims)))
+        fd_orders = kwargs.get('fd_order', tuple([expr.time_order if
```

Comment:
Note for reviewer, the `getattr` instead of `.` is necessary as in some case a dummy `_X0` dimension is the input (during solve for example) that does not have a `is_TIme` property.

Reply:
I think I'd prefer a try-except (AttributeError),then.

Said that, not sure I understand why the default fd order should be different depending on whether it's a TimeFunction or not?

Reply:
It's not different it's just picking up the order of the expression, and if derivative wrt to time you need time_order not space_order.

>I think I'd prefer a try-except (AttributeError),then.

Not sure I do will think about it

Reply:
you use getattr only if the attribute you're trying to access is a **variable** (or in general if you don't know statically what you will access) , but in this it's a constant (is_Time).

also, about this: 

> if derivative wrt to time you need time_order not space_order.

why do you say "need time_order" ? wouldn't a convention be better (eg, always default fd_order=space_order)? perhaps I'm the one missing something here

Reply:
The convention is you take the fd order of the dimension, if it's time it's time_order, which is what this fixes, I am really confused of what the issue is here. This not about wether it's a TImeFunction or a Function, it's about if it `.dt` or `.dx`.

Reply:
> This not about wether it's a TImeFunction or a Function, it's about if it .dt or .dx
ah, OK, this wasn't clear :))) my bad

Reply:
perhaps BS but...if you do

```
[ ... for d in new_dims]
```

instead of 
```
[... for d in dims]
```

then you shouldn't need the getattr, nor the try/except right....? these would be guaranteed to be Dimensions

Reply:
Not it's not guaranteed, new_dims is just a rearranged filtered version of the input wrt but it's still a Dummy if input is a Dummy.

Reply:
what is a Dummy?

Reply:
It's a sympy type, it's kinda like Symbol but unique ie `Dummy("x") == Dummy("x")` is False. Mainly sympy uses it as tmp variable in subs/solve/...

Reply:
you see, in its current form, the code doesn't make clear why we're using getattr. With a try-except, you'd be able to explain the Dummy thing as a comment in the except branch

Reply:
But doesn't really fit nicely as need an extra loop to have a try except for each dimension then go back to kwargs.... that would be pretty ugly